libvast/vast/format/ostream_writer.hpp

```diff
@@ -0,0 +1,54 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+#include <vector>
+
+#include "vast/format/writer.hpp"
+
+namespace vast::format {
+
+class ostream_writer : public writer {
+public:
+  // -- member types -----------------------------------------------------------
+
+  using ostream_ptr = std::unique_ptr<std::ostream>;
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  ostream_writer(ostream_ptr out);
+
+  ostream_writer() = default;
+
+  ostream_writer(ostream_writer&&) = default;
+
+  ostream_writer& operator=(ostream_writer&&) = default;
+
+  ~ostream_writer() override;
+
+  // -- overrides --------------------------------------------------------------
+
+  caf::expected<void> flush() override;
+
+protected:
+  // Buffer for building lines before syncing writing to out_.
+  std::vector<char> buf_;
```

Comment:
Can you add a comment at the beginning why this is necessary and what the difference to `sync_with_stdio(false)` is?

Reply:
Good point, sorry for not mentioning `sync_with_stdio` in the discussion. I've also checked `master` with `sync_with_stdio(false)` but it makes very little difference in performance. I'll add the comment.